tests/other_api/controllers/cpp_device_with_same_name/cpp_device_with_same_name.cpp

```diff
@@ -0,0 +1,22 @@
+#include <webots/Brake.hpp>
+#include <webots/Robot.hpp>
+
+#include "../../../lib/TestUtils.hpp"
+
+using namespace webots;
+
+int main(int argc, char **argv) {
+  Robot *robot = new Robot();
+  TestUtils ts(robot, argv[0]);
+
+  int deviceTag1 = robot->getDeviceTagFromIndex(0);
+  int deviceTag2 = robot->getDeviceTagFromIndex(1);
+  Brake *device1 = dynamic_cast<Brake *>(robot->getDeviceFromTag(deviceTag1));
+  Brake *device2 = dynamic_cast<Brake *>(robot->getDeviceFromTag(deviceTag2));
```

Comment:
Minor: just wondering if we could put more constness here:
```suggestion
  const Robot *robot = new Robot();
  TestUtils ts(robot, argv[0]);

  const int deviceTag1 = robot->getDeviceTagFromIndex(0);
  const int deviceTag2 = robot->getDeviceTagFromIndex(1);
  const Brake *device1 = dynamic_cast<Brake *>(robot->getDeviceFromTag(deviceTag1));
  const Brake *device2 = dynamic_cast<Brake *>(robot->getDeviceFromTag(deviceTag2));
```